src/borg/repository.py

```diff
@@ -675,6 +677,8 @@ def complete_xfer(intermediate=True):
             nonlocal unused
             # commit the new, compact, used segments
             segment = self.io.write_commit(intermediate=intermediate)
+            self.segments.setdefault(segment, 0)
+            self.compact[segment] += LoggedIO.header_fmt.size
```

Comment:
how about only doing this for intermediate commits and not registering final commits for compaction (neither here, nor in commit() method)?

that might avoid the "not any commit found, beyond repair" issue the tests are currently stumbling over.
and also it would make the repo more robust, while reducing leftover commit segment clutter.

Reply:
^ that wouldn't help against the "lots of 17-byte files issue" as the intermediate commits are within the current segment file and only the final ones are separate.